projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentService.java

```diff
@@ -106,9 +106,27 @@ private void validateFieldsCanBePatched(
         && !mergedDeployment.getGitCommitSha().equals(oldDeployment.getGitCommitSha())) {
       throw new ApiConstraintError(
           "job_version",
-          "same as current job version when using PATCH request.",
+          "same as the current job version when using PATCH request.",
           mergedDeployment.getGitCommitSha(),
-          "Use PUT HTTP request to change job version.");
+          "Use POST HTTP request to change job version.");
+    } else if (mergedDeployment.getPythonVersion() != null
+        && !mergedDeployment.getPythonVersion().equals(oldDeployment.getPythonVersion())) {
+      throw new ApiConstraintError(
+          "python_version",
+          String.format(
+              "same as the current python version -- %s -- when using PATCH request.",
+              oldDeployment.getPythonVersion()),
+          mergedDeployment.getPythonVersion(),
+          "Use POST HTTP request to change python version.");
```

Comment:
nit: Is this patch/post use case specified in documents other than the source code?

Reply:
There will be an update to the associated VEP in a separate PR.